frontend/src/metabase/admin/settings/components/widgets/SecretKeyWidget/SecretKeyWidget.tsx

```diff
@@ -7,11 +7,19 @@ import { GenerateButton, SecretKeyWidgetRoot } from "./SecretKeyWidget.styled";
 interface SecretKeyWidgetProps {
   onChange: (token: string) => void;
   setting: any;
+  confirmation?: {
```

Comment:
If we end up refactoring how the embedding version of this is setup, can we make this required in props?
```suggestion
  confirmation: {
```